crates/net/network/src/swarm.rs

```diff
@@ -172,10 +173,20 @@ where
                 return Some(SwarmEvent::TcpListenerClosed { remote_addr: address })
             }
             ListenerEvent::Incoming { stream, remote_addr } => {
-                if let Err(limit) = self.state_mut().peers_mut().on_inbound_pending_session() {
-                    // We currently don't have additional capacity to establish a session from an
-                    // incoming connection, so we drop the connection.
-                    trace!(target: "net", %limit, ?remote_addr, "Exceeded incoming connection limit; dropping connection");
+                if let Err(err) =
+                    self.state_mut().peers_mut().on_inbound_pending_session(remote_addr.ip())
+                {
+                    match err {
+                        InboundConnectionError::IpBanned => {
+                            trace!(target: "net", ?remote_addr, "The incoming ip address is in the ban list");
+                        }
+                        InboundConnectionError::ExceedsLimit(limit) => {
+                            // We currently don't have additional capacity to establish a session
+                            // from an incoming connection, so we drop
+                            // the connection.
+                            trace!(target: "net", %limit, ?remote_addr, "Exceeded incoming connection limit; dropping connection");
+                        }
+                    }
                     return None
```

Comment:
Shouldn't we be propagating the error? This seems to eat it

Reply:
I looked at doing this but it didn't make sense to me as I don't think we should be putting out a `NetworkEvent` every time we don't open a connection because an address was blacklisted. I don't think there is anything useful that can be done with this information, so propagating this into a swarm event would just cause the trace to be done at the network manager which seemed unnecessary.

Reply:
got it - makes sense unless @mattsse disagrees